ministryofjustice/correspondence_tool_staff
# CT-2223 Edit case details

## Description
This introduces the ability to edit Offender SAR cases, adding "change" links for each group of fields corresponding to the grouping on the Create Case journey

The form partials are common to both create and edit journeys, and the back link is now a helper so that it can be told which context it's to work in. In the create context, it works out the previous step from the "steps" defined for the form object. 

There's changes to the gem file and bundler version on CircleCI that I now cannot remember why I've got included in this PR, and I am not sure these are needed now that master has moved on - this is party why I want to get this merged sooner rather than later and have the tidy up work done on master, not this

## Self-review checklist

 
### Screenshots

![image](https://user-images.githubusercontent.com/1161161/80360019-a0972580-8876-11ea-8714-723f4431ad3c.png)

 
![image](https://user-images.githubusercontent.com/1161161/80360420-3b8fff80-8877-11ea-9ace-85bdaaeba246.png)

### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2223
 
### Deployment
none
 
### Manual testing instructions
Click change to edit the fields, ensuring the right form partial is displayed.
Continue should return to the case show page if you're editing. 
Form validation issues should reload the same view. Back should always take you back to the show page without saving any changes.

Need to check the case creation journey too for Offender SARs (log in as Brian Rix)
